src/app/(protected)/notebooks/page.tsx

```diff
@@ -1,14 +1,99 @@
 "use client"
 
-import React from 'react';
-import { useAuth } from '@components/auth-provider';
+import { useState } from 'react';
+import { useQuery, gql } from '@apollo/client';
+import { useRouter } from 'next/navigation'
+import Notify from '@config/notiflix-config';
+import Fuse from 'fuse.js'
+import SearchBar from '@/components/search-bar';
+import Navigation from '@/components/navigation';
+import { Spinner } from '@/components/loader'
+import NotebookList from '@/components/notebooks/cards/notebook-list'
+import { useAuth } from '@/components/auth-provider';
 
-export default function Page(): JSX.Element {
-  const { user } = useAuth();
+const GET_NOTEBOOKS_QUERY = gql`
+query GetNotebooks {
+    notebooks {
+      id
+      title
+      description
+    }
+  }
+`
+const fuseOptions = {
+  keys: [
+    "title",
+    "description"
+  ]
+}
+
+export default function Page() {
+  const router = useRouter();
+  const user = useAuth();
+  const [error, setError] = useState<boolean>(false);
+  const [notebookData, setNotebookData] = useState<Notebook[] | null>(null);
+  const [searchData, setSearchData] = useState<Notebook[] | null>(notebookData);
+
+  let { loading: notebookLoading, error: notebookError, data: queryNotebookData } = useQuery(GET_NOTEBOOKS_QUERY, {
+    onCompleted: () => {
+      setNotebookData(queryNotebookData.notebooks);
+    },
+    onError: (error) => {
+      Notify.failure(`${error.message}!`);
+      setError(true);
+    }
+  });
+
+  if (!searchData && notebookData) { setSearchData(notebookData); }
+
+  const doFilter = (query: string) => {
+    setSearchData(searchFilter(notebookData, query));
+  }
+
+  let notebookList = <Spinner className='self-center pt-8 w-[35px] h-[35px]' />;
+  if (error) {
+    notebookList = <h1 className='self-center'>Error Loading Notebooks</h1>
+  }
+  else if (searchData) {
+    notebookList = <NotebookList notebooks={searchData} />
+  }
+
+  const searchBar = <SearchBar onChange={(e) => doFilter(e.currentTarget.value)} />
 
   return (
-    <div>
-      <div>USERNAME: {user?.username}</div>
+    <div className="flex flex-col h-screen content-center">
+      <Navigation />
+      <div className='self-center flex flex-col w-10/12 lg:w-6/12 h-screen'>
+        <div className='min-h-[20px]'></div>
+        <div className="flex flex-col gap-10">
+
+          <div>
+            <h1 className='text-2xl pl-2 pb-8 font-bold'>My Notebooks</h1>
+            <div className='flex flex-row grow'>
+              {searchBar}
+              <div className='flex flex-row grow justify-end'>
+                <button type="button" className="green-button font-medium rounded-lg text-sm px-10 py-2">New</button>
+              </div>
+            </div>
+          </div>
+
+          {notebookList}
```

Comment:
I think we should handle this way differently. The way it's currently done is a bit confusing and might make it difficult to understand what's going on there. Here's what I would recommend to handle this whole thing. 

For the loading, since we don't do any other backend processing, we can simply show the spinner like on other page. Generally, that's how we want to handle loading. We can then [refetch](https://www.apollographql.com/docs/react/data/queries/#refetching) if needed but that's a conversation for another time.
```tsx
if (loading)
  return <Spinner />
```

Since we handle loading differently, the whole notebook-list component is probably no necessary anymore. By that I mean that the code could be placed here directly.

To handle a loading error, honestly, just a notification that indicates that an error happened should be fine. Returning null (or a message like you did) is fine too. That said, one thing we could do is to have a button that allows us to `refetch` the data. Maybe something that looks like that. It is not required to do this.

Reply:
fixed, didnt catch this one last time, oops